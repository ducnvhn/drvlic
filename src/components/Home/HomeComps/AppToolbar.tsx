import React from 'react'
import {
  useStyles
} from './MainHome'
import {
  AppBar,
  Toolbar,
  IconButton,
  Typography
} from '@material-ui/core'
import MenuIcon from '@material-ui/icons/Menu'
import AccountCircleIcon from '@material-ui/icons/AccountCircle'
import clsx from 'clsx'
import {
  useAuth
} from '../../../context/AuthenticationContext'
import {
  useLocation
} from 'react-router-dom'

interface IProps {
  open: boolean
  handleDrawerOpen: () => void
  logout: () => void
}


// type PMap = Map<string, string>
const paths = new Map<string,string> ()
paths.set('/home', 'Trang chủ')
paths.set('/', 'Trang chủ')
paths.set('/settings', 'Thiết lập')
paths.set('/shops', 'Cửa hàng')
paths.set('/agent', 'Nhân viên')
paths.set('/stock', 'Kho hàng')
paths.set('/products', 'Sản phẩm')
const AppToolbar:React.FC<IProps> = ({
  open,
  handleDrawerOpen,
  logout
}) => {
  const { getUser } = useAuth()
  const currentUser = getUser();
  const classes = useStyles()
  const {
    pathname
  } = useLocation()
  // console.log(pathname)
  return (
    <AppBar
        elevation={0}
        position="fixed"
        className={clsx(classes.appBar, {
          [classes.appBarShift]: open,
        })}
      >
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label="open drawer"
            onClick={handleDrawerOpen}
            edge="start"
            className={clsx(classes.menuButton, {
              [classes.hide]: open,
            })}
          >
            <MenuIcon data-testid="menuicon" />
          </IconButton>
          <Typography className={classes.toolbarTitle} variant="h6" noWrap>
            {paths.get(pathname)}
          </Typography>
          <Typography>
            {currentUser.name}
          </Typography>
          <IconButton style={{color:'white'}} onClick={() => logout()}>
            <AccountCircleIcon data-testid="accounticon" />
          </IconButton>
        </Toolbar>
      </AppBar>
  )
}
export default AppToolbar