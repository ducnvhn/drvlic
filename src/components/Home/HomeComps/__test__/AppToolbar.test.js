import React from 'react'
import {
  render,
  screen
} from '@testing-library/react'
import AppToolbar from '../AppToolbar'
import AuthProvider from '../../../../context/AuthenticationContext'

test('AppToolbar is rendered correctly', async () => {
  const comp = render(
    <AuthProvider value={{getUser: () => null}}>
      <AppToolbar />
    </AuthProvider>
  )
  const { getByText } = comp
  expect(getByText(/e-pharma/i)).toBeInTheDocument()
  expect(screen.queryByTestId('menuicon')).toBeInTheDocument()
  expect(screen.queryByTestId('accounticon')).toBeInTheDocument()
  expect(comp).toMatchSnapshot()
})