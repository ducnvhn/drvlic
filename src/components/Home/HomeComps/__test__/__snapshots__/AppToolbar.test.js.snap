// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppToolbar is rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-2 mui-fixed MuiPaper-elevation0"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-4 MuiIconButton-colorInherit MuiIconButton-edgeStart"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                data-testid="menuicon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <h6
            class="MuiTypography-root makeStyles-toolbarTitle-10 MuiTypography-h6 MuiTypography-noWrap"
          >
            E-Pharma
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1"
          />
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            style="color: white;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                data-testid="accounticon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-2 mui-fixed MuiPaper-elevation0"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="open drawer"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-4 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              data-testid="menuicon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root makeStyles-toolbarTitle-10 MuiTypography-h6 MuiTypography-noWrap"
        >
          E-Pharma
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1"
        />
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          style="color: white;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              data-testid="accounticon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
